### DISCUSSION TITLE
How to get CurrentCell value in DataGrid

### DISCUSSION THREAD

**Discussion by ShrutiJaiswal1494** [AvaloniaUI/Avalonia#6090] [Q&A]
Hi,

In WPF there is a property called `CurrentCell` which gives the current cell value of a Datagrid. In Avalonia, I do not see any such property. Is there a way to extract the cell value?

**Comment by maxkatz6** [comment]
Hi. I created an issue to port this property https://github.com/AvaloniaUI/Avalonia/issues/6095

**Reply by maxkatz6** [reply]
If somebody has time and willing to help, I would be happy to help and review

**Comment by maxkatz6** [comment]
But it's possible anyway now by traversing visual tree.
Although it's definitely ugly code:
```csharp
var focused = FocusManager.Instance.Current;
var currentRow = focused?.FindAncestorOfType<DataGridRow>() ?? myDataGrid.FindDescendantOfType<DataGridRowsPresenter>()
    .Children.OfType<DataGridRow>()
    .FirstOrDefault(r => r.FindDescendantOfType<DataGridCellsPresenter>()
        .Children.Any(p => p.Classes.Contains(":current")));
var item = currentRow?.DataContext;

var currentCell = currentRow?.FindDescendantOfType<DataGridCellsPresenter>().Children
    .OfType<DataGridCell>().FirstOrDefault(p => p.Classes.Contains(":current"));
```

**Reply by maxkatz6** [reply]
There might be better way to do so, but I haven't spent much time on it
